File: 프로그래머스/lv1/42748. K번째수/K번째수.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
	vector<int> answer;

	for (int i = 0; i < commands.size(); i++) {
		vector<int> arr;
		arr.assign(array.begin() + commands[i][0]-1, array.begin()+commands[i][1]);
		sort(arr.begin(), arr.end());
		answer.push_back(arr[commands[i][2]-1]);
	}
	return answer;
}

File: 프로그래머스/lv1/42748. K번째수/K번째수.java
import java.util.*;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        for(int cmd=0; cmd<commands.length; cmd++){
            int k = commands[cmd][2];
            int[] n = new int[commands[cmd][1]-commands[cmd][0]+1];
            for(int i=commands[cmd][0]-1; i<commands[cmd][1]; i++){
                n[i-commands[cmd][0]+1] = array[i];
            }          
            Arrays.sort(n);
            System.out.println(n[0]);
      
           answer[cmd] = n[k-1];
            
        }
        return answer;
    }
}